data-prepper-plugins/common/src/main/java/org/opensearch/dataprepper/plugins/metricpublisher/MicrometerMetricPublisher.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.metricpublisher;
+
+import org.opensearch.dataprepper.metrics.PluginMetrics;
+import software.amazon.awssdk.core.metrics.CoreMetric;
+import software.amazon.awssdk.metrics.MetricCollection;
+import software.amazon.awssdk.metrics.MetricPublisher;
+import software.amazon.awssdk.metrics.SdkMetric;
+
+import java.time.Duration;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+public class MicrometerMetricPublisher implements MetricPublisher {
+    private static final String SERVICE_ID = "ServiceId";
+    private static final String OPERATION_NAME = "OperationName";
+    private static final Set<SdkMetric<String>> DEFAULT_DIMENSIONS = Set.of(CoreMetric.SERVICE_ID, CoreMetric.OPERATION_NAME);
+
+    private final PluginMetrics pluginMetrics;
+
+    public MicrometerMetricPublisher(final PluginMetrics pluginMetrics) {
+        this.pluginMetrics = pluginMetrics;
+    }
+
+    @Override
+    public void publish(final MetricCollection metricCollection) {
+        final Map<String, String> dimensions = metricCollection.stream()
+                .filter(metricRecord -> DEFAULT_DIMENSIONS.contains(metricRecord.metric()))
+                .collect(Collectors.toMap(metricRecord -> metricRecord.metric().name(), metricRecord -> metricRecord.value().toString()));
+
+        addMetricsToPluginMetrics(metricCollection, dimensions);
+        metricCollection.children().forEach(
+                child -> addMetricsToPluginMetrics(child, dimensions)
```

Comment:
Can child collection have grandchild collections and so on? Should we put these two lines inside `addMetricsToPluginMetrics` so it runs recursively?

Reply:
Yes, good catch. I will fix this.